lib/stdio/encoding.js

```diff
@@ -7,8 +7,16 @@ export const handleStreamsEncoding = (stdioStreams, {encoding}, isSync) => {
 		return stdioStreams.map(stdioStream => ({...stdioStream, encoding}));
 	}
 
-	const value = encodingEndGenerator.bind(undefined, encoding);
-	return [...stdioStreams, {...stdioStreams[0], type: 'generator', value, encoding: 'buffer'}];
+	const transform = encodingEndGenerator.bind(undefined, encoding);
+	return [
+		...stdioStreams,
+		{
+			...stdioStreams[0],
+			type: 'generator',
+			value: {transform, binary: true},
```

Comment:
This is the transform that we internally use for the `encoding: 'hex | base64'` option. It does not need to operate line-wise, so we make opt-out since this is more efficient.